### PR TITLE
TECH-7764 change user fields to use full name

### PR DESCRIPTION
**Pull request by eric38383**
[TECH-7764 change user fields to use full name](https://industrydive.atlassian.net/browse/TECH-7764)

### Non-technical Context

This merge will displayed the User's first and last name in the Interaction and Sources Admin instead of the User's username.

### Technical Context

- Create custom admin methods for displaying created by field. 
- Change interviewers_listview to use full name. 
- Overwrite form field for many to many method in Interaction Admin.
- Create InterviewerChoice class to overwrite label from instance to display user's full name.

### Reproduction Steps

- [x] Login to admin for sourcedive.
- [x] Go to source list view. Created by column should display username.
- [x] Go to sources change page and open advanced info. Created by field should display username. Also attempt to add a new interaction on the same page below. Interviewer selects should display username.
- [x] Go to interaction list view. Created by and interviewer(s) column should display username.
- [x] Go to interaction add page. Select dropdown for interviewer(s) should display username.
- [x] Go to interaction change page. Repeat above step. Created by field should also display username.

### Manual Testing Instructions

Repeat the reproduction steps, except User's first and last name should appear instead of user' s username.

- [x] Login to admin for sourcedive.
- [x] Go to source list view. Created by column should display user's first and last name.
- [x] Go to sources change page and open advanced info. Created by field should display user's first and last name. Also attempt to add a new interaction on the same page below. Interviewer selects should display user's first and last name.
- [x] Go to interaction list view. Created by and interviewer(s) column should display user's first and last name.
- [x] Go to interaction add page. Select dropdown for interviewer(s) should display user's first and last name.
- [x] Go to interaction change page. Repeat above step. Created by field should also display user's first and last name.
- [x] Go to Dive add and change pages. User select forms should display user's first name and last name.
- [x] If user doesn't have name, form will display username as a default

#### Developer

- [X] Ensure code complies with the [style guide](https://industrydive.atlassian.net/wiki/spaces/TECH/pages/639500407/Tech+Team+s+Style+Guides)
- [X] Devise edge cases to test the bug fix or feature being merged
- [X] Make sure the code is clean and understandable
- [X] Ensure code is not overly inefficient
- [X] Ensure documentation is understandable and accurate, including:
    - [X] Code comments and doc strings
    - [X] Technical documentation

#### Reviewer

- [x] Ensure code complies with the [style guide](https://industrydive.atlassian.net/wiki/spaces/TECH/pages/639500407/Tech+Team+s+Style+Guides)
    - If there are few or minor issues, add inline GitHub comments
    - If there are many issues or it seems like the developer did not follow the style guide, leave a comment asking them to do so
- [x] Devise edge cases to test the bug fix or feature being merged
- [x] Make sure the code is clean and understandable
    - [x] Ask for inline comments on unclear sections
- [x] Ensure code is not overly inefficient
- [x] Ensure documentation is understandable and accurate, including:
    - [x] Code comments and doc strings
    - [na] Technical documentation

### CODE REVIEW

**Comment by greglinch** [comment]
@eric38383 Thanks for doing this! One quick update needed, which is my fault for not specifying in the ticket description, is that we still need `get_username()` as a fallback if there's no value for `get_full_name()`. 

Also, I got this on the interaction list view:

Screen Shot 2020-03-09 at 3 56 42 PM

**Comment by eric38383** [comment]
Ok cool, will take a look at that!

**Comment by eric38383** [comment]
@greglinch Wasn't able to duplicate your error. I probably need to fill up my db a little more with some stuff. will investigate further when i get chance

**Comment by greglinch** [comment]
@eric38383 Ah, ok. I just shared a Google Drive folder with data you can load to your local version.

**Comment by eric38383** [comment]
@greglinch Updated Dive add and change pages well. Created fallback if full name is missing on all forms. Created a helper function as well, not sure how you prefer to organize that, I could use some guidance.

On the error, it seems that `Interaction.objects.get(pk=4)`  has a null for created_by field. Let me know how best to proceed here.

File: sources/admin.py [hunk 1]
```diff
@@ -16,12 +19,33 @@
 )
 
 
+def get_username(obj):
```
Comment (greglinch) [inline sources/admin.py]: How about this for clarity?
```suggestion
def get_user_display_name(obj):
```

File: sources/admin.py [hunk 2]
```diff
@@ -16,12 +19,33 @@
...
+    return obj.get_full_name() or obj.username
+
+class UserChoiceField(ModelMultipleChoiceField):
+    """
+        This will change how the user (might be field interviewer, etc) is
+        displayed inside the dropdowns. It will now show the user's full name
+        instead of the user's username.
+    """
+    def label_from_instance(self, obj):
+        return get_username(obj)
```
Comment (greglinch) [inline sources/admin.py]:
```suggestion
        return get_user_display_name(obj)
```

File: sources/admin.py [hunk 3]
```diff
@@ -162,10 +191,26 @@ def save_model(self, request, obj, form, change):
...
+            This will display a custom created_by field by using
+            the user's first and last name instead of the user's username
+        """
+        return get_username(obj.created_by)
+    get_created_by.short_description = 'Created By'
 
     def interviewers_listview(self, obj):
         interviewers_list = obj.interviewer.all()
-        interviewers = [interviewer.username for interviewer in interviewers_list]
+        interviewers = [get_username(interviewer) for interviewer in interviewers_list]
```
Comment (greglinch) [inline sources/admin.py]:
```suggestion
        interviewers = [get_user_display_name(interviewer) for interviewer in interviewers_list]
```

File: sources/admin.py [hunk 4]
```diff
@@ -480,6 +525,14 @@ def save_model(self, request, obj, form, change):
         ## save
         super(PersonAdmin, self).save_model(request, obj, form, change)
 
+    def get_created_by(self, obj):
+        """
+            This will display a custom created_by field by using
+            the user's first and last name instead of the user's username
+        """
+        return get_username(obj.created_by)
```
Comment (greglinch) [inline sources/admin.py]:
```suggestion
        return get_user_display_name(obj.created_by)
```

**Review by greglinch — CHANGES_REQUESTED** [review]
@eric38383 Thanks for the updates! I fixed my local db to populate that field, so I'm all good now on errors.

**Comment by eric38383** [comment]
updated with new function name

**Comment by greglinch** [comment]
@eric38383 Sorry it took so long to approve this. All good!
